trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/TrinoS3ProxyClient.java

```diff
@@ -84,6 +87,11 @@ public void proxyRequest(SigningMetadata signingMetadata, ParsedS3Request reques
     {
         URI remoteUri = remoteS3Facade.buildEndpoint(UriBuilder.newInstance(), request.keyInBucket(), request.bucketName(), signingMetadata.region());
 
+        // TODO log/expose any securityController error
+        if (!securityController.apply(request, signingMetadata.region(), request.rawQuery()).canProceed()) {
```

Comment:
`rawQuery` is also in `request`

Also, on a similar note to my previous comment - https://github.com/trinodb/s3-proxy/pull/54#discussion_r1636495979

Should we pass down the identity we have in `signingMetadata`? We have already parsed the access key ID, why not give it to the security controller? It can apply more complex logic if it wants, but the identity of the caller seems like a reasonable thing to give to the security plugin by default